syntax = "proto3";

package ovms_ipc;

import "ie_serving/messaging/data_attributes.proto";

message EndpointRequest{
    oneof type{
        PredictRequest predict_request = 1;
        ModelMetadataRequest model_metadata_request = 2;
        ModelStatusRequest model_status_request = 3;
    }
}

message PredictRequest{
    message Data{
        oneof attributes{
            NumpyAttributes numpy_attributes = 1;
            StringAttributes string_attributes = 2;
            BinaryAttributes binary_attributes = 3;
        }
        string input_name = 4;
        string shm_name = 5;
    }
    repeated Data inputs = 1;
    string return_socket_name = 2;
}

message ModelMetadataRequest{
    int32 requested_version = 1;
}

message ModelStatusRequest{
    int32 requested_version = 1;
}
